test/javascript/transpile-test.ts

```diff
@@ -210,11 +210,11 @@ describe("transpileModule(input, root, path)", () => {
   it("rewrites npm imports", async () => {
     const input = 'import "npm:d3-array";';
     const output = (await transpileModule(input, options)).split("\n").pop()!;
-    assert.strictEqual(output, 'import "../_npm/d3-array@3.2.4/_esm.js";');
+    assert.strictEqual(output, 'import "../_npm/d3-array@3.2.4/_esm.js"/* observablehq-file */;');
```

Comment:
If this link is tagged as `observable-file`, then we should also rewrite it to tag Observable files in it. If we don't rewrite it, I don't think we should tag it.

Reply:
I was initially doing that (see line 307 in https://github.com/observablehq/framework/pull/1731/commits/6cfb74de17c13d54608a14dd396bf56a97b30e2f#diff-a479e6903ca67c840829983998e33849155911842cf08fa118b42b2d88fe3c9b), then I think we decided to tag everything and let the server decide what it should do with it.

Reply:
My concern is that this is a reference to an `_npm` file that is tagged, but references to `_npm` files from other `_npm` files are not tagged. As a concrete example, in the docs build there is this chain:

1. `dist/chart.js` tags its import to the `_import` file:
2. `docs/.observablehq/dist/_import/chart.95e875cf.js` tags its import of an `_npm` file:
3. `docs/.observablehq/dist/_npm/@observablehq/plot@0.6.16/e828d8c8.js` does not tag its import of the `_npm` file:
4. `../../isoformat@0.2.1/c68fbd73.js`

It would be helpful if the tagging was done based on the taget of the URL, not the place the URL occurs. As it stands, the server can't really make the decision fully, because links to `_npm` files aren't consistent tagged/not tagged, so it _has_ to special case them as not needing extra authentication.